lib/minitest/mock.rb

```diff
@@ -113,8 +114,11 @@ def verify
 
     def method_missing(sym, *args, &block) # :nodoc:
       unless @expected_calls.key?(sym) then
-        raise NoMethodError, "unmocked method %p, expected one of %p" %
-          [sym, @expected_calls.keys.sort_by(&:to_s)]
+        if @delegator
+          delegate_method(sym, *args, &block)
+        else
+          raise_unmocked_method_error(sym)
+        end
```

Comment:
The part where it raises an exception when there is no delegator, was the existing behavior, nothing changed there.

As far as the other branch of the conditional, you're right, it is missing a `return` statement. It is supposed to either return whatever the response is from the delegator or raise if the method is not found there as well (maybe the message could be different in this case).